ft-sdk/src/auth/mod.rs

```diff
@@ -73,29 +75,48 @@ pub fn session_providers() -> Vec<String> {
     todo!()
 }
 
+/// Fetches user data based on a given session cookie.
+///
+/// This function fetches user data based on a given session cookie if the
+/// session cookie is found and is valid and user data is found.
+/// If the session cookie not found, it returns `None`.
 #[cfg(feature = "field-extractors")]
 pub fn ud(
     cookie: ft_sdk::Cookie<SESSION_KEY>,
     conn: &mut ft_sdk::Connection,
 ) -> Result<Option<ft_sys::UserData>, UserDataError> {
-    if let Some(v) = ft_sys::env::var("DEBUG_LOGGED_IN".to_string()) {
-        let mut v = v.splitn(4, ' ');
-        return Ok(Some(ft_sys::UserData {
-            id: v.next().unwrap().parse().unwrap(),
-            identity: v.next().unwrap_or_default().to_string(),
-            name: v.next().map(|v| v.to_string()).unwrap_or_default(),
-            email: v.next().map(|v| v.to_string()).unwrap_or_default(),
-            verified_email: true,
-        }));
-    }
```

Comment:
This is wrong. Now `ud()` will return `None` if `DEBUG_LOGGED_IN` is set but there is session cookie. Earlier it used to return user corresponding to `DEBUG_LOGGED_IN`.

I am not convinced we need two functions, if you are creating session id type, then do not accept cookie, and crate a `From` impl that converts `ft_sdk::Cookie<SESSION_KEY>` to `ft_sdk::auth::SessionID`.

Reply:
done